src/api/cpp/cvc5.cpp

```diff
@@ -5527,9 +5527,11 @@ void Solver::ensureWellFormedTerm(const Term& t) const
     bool wasShadow = false;
     if (internal::expr::hasFreeOrShadowedVar(*t.d_node, wasShadow))
     {
+      std::unordered_set<internal::Node> fvs;
```

Comment:
This block is triggered also for shadowed variables, can you preserve that behavior here?

Reply:
`wasShadow` may be set by hasFreeOrShadowedVar.

Reply:
Right. I missed the it was passed by reference here. 
The change happens here. 
https://github.com/cvc5/cvc5/blob/main/src/expr/node_algorithm.cpp#L351